src/core/control/Control.cpp

```diff
@@ -1486,33 +1486,52 @@ void Control::replaceDocument(std::unique_ptr<Document> doc, int scrollToPage) {
 }
 
 void Control::openXoppFile(fs::path filepath, int scrollToPage, std::function<void(bool)> callback) {
-    LoadHandler loadHandler;
-    std::unique_ptr<Document> doc(loadHandler.loadDocument(filepath));
+    std::unique_ptr<Document> doc{};
+    std::optional<MissingPdfData> missingPdf{};
+    int fileVersion{};
+    std::string errorMessages{};
 
-    if (!doc) {
-        std::string msg = FS(_F("Error opening file \"{1}\". Error message:{2}") % filepath.u8string() %
-                             loadHandler.getErrorMessages());
+    try {
+        LoadHandler loadHandler;
+        doc = loadHandler.loadDocument(filepath);
+
+        if (!loadHandler.getMissingPdfFilename().empty() || loadHandler.isAttachedPdfMissing()) {
+            missingPdf = {loadHandler.isAttachedPdfMissing(), loadHandler.getMissingPdfFilename()};
+        }
+        fileVersion = loadHandler.getFileVersion();
+        errorMessages = loadHandler.getErrorMessages();
```

Comment:
If we move this `getErrorMessages` call inside the `catch` block, we could print the errors that happen before the exception.

Reply:
The `loadHandler` doesn't exist anymore in the catch block. I now realize that if an exception occurs, the `errorMessages` will be left empty. However I believe that if an exception occurs, preceding warnings are likely to be unrelated. 
What I could do is have the load handler append error messages directly to a string owned by the caller. Thoughts?

Reply:
One option could move the instantiation of `loadHandler` out side of the `try-catch` so it is accessible from the `catch` block. I understand that maybe it is sub-optimal, but that would allow you to get the `errorMessages`.
It is true that if the `LoadHandler` raised an exception,  `errorMessages` may be unrelated but it may also be a hint of what was wrong before the exception too.

Reply:
The idea was to directly throw out of the `LoadHandler` when an exception occurs. I've added a commit that solves this issue as I described above.

Reply:
There are several ways, dealing with that. But first, are those error messages sometimes the reason, why the exception occurred? Or are they unrelated? 

If they may be related, 
 - we could give the Loadhandler a reference to a `vector<string>` which is filled with the error messages.
 - double catch and rethrow, handle error messages in the inner try-catch
 - fail on the first error and pass the message via an exception.
 - collect exceptions in a container and generate a message from those.